tkey/src/main/java/com/web3auth/tkey/ThresholdKey/Modules/TSSModule.java

```diff
@@ -0,0 +1,472 @@
+        String result = jniGetExtendedVerifier(thresholdKey, error);
+        if (error.code != 0) {
+            throw error;
+        }
+        return result;
+    }
+
+    public static void setTSSTag(ThresholdKey thresholdKey, String tssTag, ThresholdKeyCallback<Boolean> callback) {
+        thresholdKey.executor.execute(() -> {
+            try {
+                Result<Boolean> result = setTSSTag(thresholdKey, tssTag);
+                callback.onComplete(result);
+            } catch (Exception e) {
+                Result<Boolean> error = new Result.Error<>(e);
+                callback.onComplete(error);
+            }
+        });
+    }
+
+    private static Result<Boolean> setTSSTag(ThresholdKey thresholdKey, String tssTag) {
+        try {
+            RuntimeError error = new RuntimeError();
+            jniTSSModuleSetTSSTag(thresholdKey, tssTag, error);
+            if (error.code != 0) {
+                throw new Exception(error);
+            }
+            return new Result.Success<>(true);
+        } catch (Exception e) {
+            return new Result.Error<>(e);
+        }
+    }
+
+    public static String getTSSTag(ThresholdKey thresholdKey) throws RuntimeError {
+        RuntimeError error = new RuntimeError();
+        String result = jniTSSModuleGetTSSTag(thresholdKey, error);
+        if (error.code != 0) {
+            throw error;
+        }
+        return result;
+    }
+
+    public static GetTSSPubKeyResult getTssPubAddress(ThresholdKey thresholdKey, String tssTag, String nonce, NodeDetails nodeDetails, TorusUtils torusUtils) throws Exception, RuntimeError {
+        String extendedVerifierId = getExtendedVerifier(thresholdKey);
+        String[] split = extendedVerifierId.split("\u001c");
+        String extendedVerifierIdFormatted = split[1] + "\u0015" + tssTag + "\u0016" + nonce;
+
+        VerifierArgs verifierArgs = new VerifierArgs(
+            split[0],
+            split[1],
+            extendedVerifierIdFormatted
+        );
+        TorusPublicKey result = torusUtils.getPublicAddress(nodeDetails.getTorusNodeEndpoints(), nodeDetails.getTorusNodePub(), verifierArgs).get();
+
+        String x = result.getFinalKeyData().getX();
+        String y = result.getFinalKeyData().getY();
+        List<BigInteger> nodeIndexes = result.getNodesData().getNodeIndexes();
+        GetTSSPubKeyResult.Point pubKey = new GetTSSPubKeyResult.Point(x, y);
+        return new GetTSSPubKeyResult(pubKey, nodeIndexes);
+    }
+
+    public static Pair<String, String> getTSSShare(ThresholdKey thresholdKey, String TSSTag, String factorKey, int threshold) throws RuntimeError {
+        if(factorKey.length() > 64) {
+            throw new RuntimeException("Invalid factor Key");
+        }
+        setTSSTag(thresholdKey, TSSTag);
+
+        RuntimeError error = new RuntimeError();
+        String result = jniTSSModuleGetTSSShare(thresholdKey, factorKey, threshold, thresholdKey.curveN, error);
```

Comment:
If we use the curveN for thresholdkey here, we don't need it as a variable in the module.